analysis/cross_partition_combiners.py

```diff
@@ -258,22 +258,22 @@ def _average_utility_report(report: metrics.UtilityReport,
                             public_partitions: bool,
                             sums_actual: Tuple) -> None:
     """Averages fields of the 'report' across partitions."""
+    if not report.metric_errors:
+        return
     partitions = report.partitions_info
     if public_partitions:
         num_output_partitions = partitions.num_dataset_partitions + partitions.num_empty_partitions
     else:
         num_output_partitions = partitions.kept_partitions.mean
-    _multiply_float_dataclasses_field(report.partitions_info,
-                                      1.0 / num_output_partitions)
-    if report.metric_errors:
```

Comment:
Why did this line change?

Reply:
I've moved this check up for simplifying the code. Now, if `metric_errors` are empty then this function is no-op, having that in the beginning makes it more clear